src/main/java/Duke.java

```diff
@@ -1,10 +1,278 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
+    private static final String SAVE_FILE_PATH = "saveFile.txt";
+    private static final File SAVE_FILE = new File(SAVE_FILE_PATH);
+    private static final String DIVIDER = "____________________________________________________________";
+    private static final Scanner SCANNER = new Scanner(System.in);
+    private static final ArrayList<Task> STORED_TASKS = new ArrayList<Task>();
+
     public static void main(String[] args) {
+        loadDataFromFile();
+        welcomeMsg();
+
+        String command;
+        do {
+            command = getCommandFromUser();
+            handleCommand(command);
+        } while (!command.equals("bye"));
+
+        SCANNER.close();
+    }
+
+    public static void loadDataFromFile() {
+        try {
+            Scanner saveFileScanner = new Scanner(SAVE_FILE);
+            while (saveFileScanner.hasNextLine()) {
+                String[] taskInfo = saveFileScanner.nextLine().split(" \\| ");
+                Task currentTask;
+                if (taskInfo[0].equals("T")) {
+                    currentTask = new Todo(taskInfo[2]);
+                    if (taskInfo[1].equals("1")) {
+                        currentTask.markAsDone();
+                    }
+                } else if (taskInfo[0].equals("D")) {
+                    currentTask = new Deadline(taskInfo[2], taskInfo[3]);
+                    if (taskInfo[1].equals("1")) {
+                        currentTask.markAsDone();
+                    }
+                } else {
+                    currentTask = new Event(taskInfo[2], taskInfo[3]);
+                    if (taskInfo[1].equals("1")) {
+                        currentTask.markAsDone();
+                    }
+                }
+                STORED_TASKS.add(currentTask);
+            }
+            saveFileScanner.close();
```

Comment:
Sort of an indicator that SLAP may be violated here. You can try to extract out the stuff inside the if...else ladder to another method.
Also how `taskInfo[1]` is treated could be confusing to the one who embarks on maintaining your code.